heltour/tournament/models.py

```diff
@@ -693,7 +695,11 @@ def update_profile(self, user_meta):
 
     @classmethod
     def get_or_create(cls, lichess_username):
-        player, _ = Player.objects.get_or_create(lichess_username__iexact=lichess_username, defaults={'lichess_username': lichess_username})
+        try:
+            user = User.objects.get(username__iexact=lichess_username)
+        except User.DoesNotExist:
+            raise f'no user with lichess username {lichess_username} exists'
```

Comment:
I don't expect a `get_or_create` method to raise an error when something doesn't exist. I expect it to create that object.  I haven't analyzed where this method is used well enough to know if that makes sense everywhere, but that's what I would expect from a method named like this.

Reply:
It's a constraint violation.  Don't we expect a constraint violation in a method called create to throw an exception

Reply:
I don't see it as a constraint violation. I see it as a request to ensure that a Player with that username exists in the system. If the player doesn't exist, we create it. If the Player requires that the user also exists, then you should create both.